lib/ansible/template/__init__.py

```diff
@@ -829,7 +829,7 @@ def _fail_lookup(self, name, *args, **kwargs):
     def _now_datetime(self, utc=False, fmt=None):
         '''jinja2 global function to return current datetime, potentially formatted via strftime'''
         if utc:
-            now = datetime.datetime.utcnow()
+            now = datetime.datetime.now(datetime.timezone.utc)
```

Comment:
As much as I'd like to see this default to using an aware datetime, I'm not sure that's a good idea. This changes the behavior of `{{ now(utc=true) }}` in templates:

`2023-07-26 00:58:34.630109` (before)
`2023-07-26 00:58:34.630109+00:00` (after)

That could be a breaking change any place it's used as a string, or compared to a naive datetime object sourced from somewhere else.